source/run-command/run-command.rst

```diff
@@ -0,0 +1,185 @@
+===========
+Run Command
+===========
+
+:Title: Run Command
+:Status: Implementing
+:Minimum Server Version: N/A
+
+.. contents::
+
+--------
+
+Abstract
+========
+This specification defines requirements and behaviors for drivers' run command and related APIs.
+
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in `RFC 2119
+<https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Specification
+=============
+
+-----
+Terms
+-----
+Command
+    A structure representing a BSON document that has a shape matching a supported MongoDB operation.
+
+---------------------------
+Implementation requirements
+---------------------------
+
+All drivers MAY offer the operations defined in the following sections.
+This does not preclude a driver from offering more.
+
+--------------
+``runCommand``
+--------------
+
+The following represents how a runCommand API SHOULD be exposed.
+
+.. code:: typescript
+
+    interface Db {
+      /**
+       * Takes an argument representing an arbitrary BSON document and executes it against the server.
+       */
+      runCommand(command: BSONDocument, options: RunCommandOptions): BSONDocument;
+    }
+
+    interface RunCommandOptions {
+      /**
+       * An optional readPreference setting to apply to server selection logic.
+       * This value MUST be applied to the command document as the $readPreference global command argument if not set to primary.
+       *
+       * @defaultValue primary
+       *
+       * @see https://www.mongodb.com/docs/manual/core/read-preference/
+       */
+      readPreference?: ReadPreference;
+
+      /**
+       * An optional explicit client session.
+       * The associated logical session id (`lsid`) MUST be applied to the command.
+       *
+       * @see https://www.mongodb.com/docs/manual/core/read-isolation-consistency-recency/#std-label-sessions
+       */
+      session?: ClientSession;
+
+      /**
+       * An optional timeout option to govern the amount of time that a single operation can execute before control is returned to the user.
+       * This timeout applies to all of the work done to execute the operation, including but not limited to server selection, connection checkout, and server-side execution.
+       *
+       * @ see https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst
+       */
+      timeoutMS?: number;
+    }
+
+Deviations
+^^^^^^^^^^
+
+* Drivers MAY rename their API to match the language syntax.
```

Comment:
You can optionally refer to the deviations guidelines in the CRUD spec, since that's much more fleshed out. Feel free to disregard, though.

Separate note: "Deviations" is typically a top-level field for the spec itself, so I think this can be moved up and out of the API section.

Reply:
Moved deviations up, and linked to the CRUD spec I think it has more detail worth referring to